bazel/scripts/lint.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env sh
+echo "running lint script"
+
+#Golang setup
+export PATH="${PWD}/${GO_LOCATION}/bin:$PATH"
+export SOURCE_LINT="$(find $PWD -name ${GO_LIBRARY_NAME})"
+export GOPATH=$SOURCE_LINT
+
+echo "running lint on directory $(find $PWD -name enfabrica)/enkit"
+
+# this is necessary for cache + homdir lib errors
+export HOME="$PWD"
+mkdir -p .cache
+
+
+#create output files
+mkdir -p $(dirname ${LINT_OUTPUT})
+touch ${LINT_OUTPUT}
+export LINT_OUTPUT="$PWD/${LINT_OUTPUT}"
+
+cd $(find $PWD -name enfabrica)/enkit
+echo "$(golangci-lint run --path-prefix / --issues-exit-code 0)" >> ${LINT_OUTPUT}
```

Comment:
`golangci-lint` is a go program. Most of them are very easy to bring into a bazel hermetic environment, without having to rely on the tool being installed in the system (eg, bazel will download and build it for you - bazel will even version it for you! fixing most of the reasons why the golangci-lint guys don't want you to build it!).

How?
- (first time, start from a clean branch, I recommend you send a separate PR for this the first time)
- from the top level directory of the repository, run `go get ...whatever...golangci-lint`, it will change the `go.mod` file, versioning the tool in our repository.
- run something like `bazelisk run //:gazelle -- update-repos -from_file=go.mod -to_macro=bazel/go_repositories.bzl%go_repositories` - this will version all the dependencies in bazel. This would normally update the WORKSPACE file, but in the enkit repo, WORKSPACE file just imports bazel/go_repositories.bzl, and runs the go_repositories macro, thus the extra argument.
- this will generate a BUILD.bazel file for the golangci-lint tool.

Once this is done, you can have a tool dependency in the rule, pointing to the built golangci-lint tool, and pass the path as an environment variable, for example.

In the end, what will happen is that when your rule is run, bazel will look for the tool, won't find it, will download and build it (at the specific version *we* tested and coded in the workspace file, not a random version on the guy's laptop), and run it (and cache it).

It looks like a lot of steps, but once you know how to do it and you've done it a couple times, it's like 10-15 mins work.